src/net_processing.cpp

```diff
@@ -3274,39 +3274,16 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             m_num_preferred_download_peers += state->fPreferredDownload;
         }
 
-        // Self advertisement & GETADDR logic
         if (!pfrom.IsInboundConn() && SetupAddressRelay(pfrom, *peer)) {
```

Comment:
1. `SetupAddressRelay` used to have a misleading comment: "First addr message we have received from the peer, initialize m_addr_known". We were not receiving any addr messages from the peer here, yet we initialized it. It seems like this was the case from the beginning.
2. After this PR, calling `SetupAddressRelay` here seems unnecessary. If we drop it, the comment becomes valid :)
Although the inconsistency is a good signal the comment is confusing, so perhaps we should drop it. Not very useful anyway, at least the current code is pretty readable.

Reply:
So yeah: 1) Drop `SetupAddressRelay` here; 2) Possibly drop the useless comment from `SetupAddressRelay`?

Reply:
Alright, this has side-effects. `MaybeSendAddr` (another self-ad) will stop working until we get an ADDR message from the peer (not the case before the change i suggest here).

Perhaps this code should be:
```
if (!pfrom.IsInboundConn()) {
    SetupAddressRelay(pfrom, *peer);
    ....
}

Reply:
I think the current code is correct, `SetupAddressRelay` here has three tasks:
1) Initialize `m_addr_known` for outbound connections (unless block-relay-only), see #21528
2) Prevent Self-Advertisements to block-relay-only outbound peers (for which `SetupAddressRelay` returns false)
3) Prevent a GETADDR being sent to block-relay-only outbound peers

Since 1) and 3) are still valid after this PR, I think we both need to keep calling `SetupAddressRelay` and skip if it returns false.
I think this should be documented better (had to look up 21528 to check how it worked), so I will add some more documentation for this soon.

Reply:
The refactoring I suggested in my last message here is indeed wrong, because it ignores the returned value.

Do you agree that the comment inside `SetupAddressRelay` is wrong though? That's the only thing that remains worrying me.

Reply:
> Do you agree that the comment inside `SetupAddressRelay` is wrong though?

Yeah, it's incomplete because it only applies to inbound peers. I added documentation there and before the call to `SetupAddressRelay` during version processing (as a second commit, because it's not directly related to the changes).

Reply:
nit: "Prevent GETADDR from being sent" sounds very complicated, and also might have additional unnecessary meaning of pro-actively preventing something.

I'd say "Don't send GETADDR [...]" is better. Feel free to ignore.

Reply:
missed this comment, changed wording now.

Reply:
I agree that it this would be better to understand if it's a bit more spread out, so I took your suggestion and changed the comment - hopefully it's clear now?